product_properties/README.md

```diff
@@ -0,0 +1,8 @@
+Product properties
+==================
+
+YAML files contained here are processed in lexicographic order, and they allow to define product properties in a efficient way.
+Processing of those files can use `jinja2`, and macros or conditionals have access to product properties defined previously and in the `product.yml`.
+Properties read from these yaml files can be overriden by properties from files processed later, while the properties specified in the `product.yml` override all previous definitions.
+
+Conventionally, use the filename prefix `10-` to define defaults, and use subsequent prefixes to override those defaults based on products.
```

Comment:
I don't like it, I would prefer to have a single source of truth, i.e. define a given option only in 1 file.

Reply:
This is tricky: [YAML requires](https://yaml.org/spec/1.2-old/spec.html#mapping) that in mappings, keys have to be unique, so setting a value to the default and redefining it later isn't legal, although libraries may support it.
The idea to define everything in one file is reasonable to me, so here are the options:

What is illegal:
```
key: value
key: another-value
```

Dedicated default section:
```
default:
  key: value
overrides:
  key: another-value
```
while the build system has to write overrides over defaults.

List of mappings:
```
- key: value
- key: another-value
```
while the build system has to assemble the set of key-value mappings gradually by iterating over the list.

Reply:
I think that the first option can be unreliable and the third option can be unexpected by newcomers so I prefer the second option because it's explicit. But let's take a look how it fits into the whole context of the "product properties" feature.

Reply:
What are the benefits of having the key overrides in centrally placed lexicographically ordered files?
Why not keep each products properties in its own `product.yml`?
This also facilitates the removal of a product.

Reply:
> What are the benefits of having the key overrides in centrally placed lexicographically ordered files? Why not keep a centrally defined default, and each product overrides properties in its own `product.yml`, or maybe a product property file in the product directory, like `products/rhel8/product_properties.yml`? This also facilitates the addition and removal of a product. For example, one can copy the file with the default values, and tweak / remove keys.

Sorry, race condition. The override possibility is no longer considered, and if we introduce core product attributes s.a. "family" (ubuntu, debian, ol, rhel) and major version (8, 2204), we may even make the core product properties available at other properties definition time, which would take the ordering away completely.